bundles/org.openhab.binding.tplinksmarthome/src/main/resources/OH-INF/thing/channels.xml

```diff
@@ -21,27 +21,27 @@
 
 	<!-- Absolute Temperature Channel -->
 	<channel-type id="colorTemperatureAbs1" advanced="true">
-		<item-type>Number</item-type>
+		<item-type unitHint="K">Number:Temperature</item-type>
```

Comment:
AFAIK the unitHint is not needed if the pattern is forcing the unit. 
The unitHint is only meaningfull if the pattern is %unit%

Reply:
I don't believe it is correct.
Main UI will propose the correct unit only if set by unitHint.

Reply:
I wonder what happens if the unit in the state pattern and the chosen (based on the unitHint) unit conflict. But you might be right as i based this on the dimenionless channel for percentages and that is a little different anyway.

Reply:
It should just set the pattern and the unit separately. If the unit is mirek, and the pattern is K, then the item will store the state in mireks, but the UI should show the state in K. Though this configuration might confuse min/max in the state description, since it's only aware of the pattern, and will assume that's the unit for min/max.

Reply:
See this for info.. 
https://github.com/openhab/openhab-core/issues/4432